### DISCUSSION TITLE
OSS and CSS: The Issues that Plague the Digital Landscape

### DISCUSSION THREAD

**Discussion by jason-muljadi** [albertkun/21S-ASIAAM-191A#106] [Past Thinking Caps]
Open source and closed source are two well-researched but still very misunderstood concepts that continually change due to the advances in the field of technology. The arguments for and against open sources were prevalent starting in the 2000s, and, and the premises laid the foundations for the potentiation of open-source software, which provides both a moralistic ground to the sharing of information in the digital landscape.

### Part 1: Open-Source, Open Possibilities
The internet has been a massively successful tool that has allowed for creative innovation in all aspects of personal lives and careers. It is with open-source software (OSS) that databases can be more easily accessible and this allows for more collaboration, especially in the field of research. Krogh and Hippel's article shows how different sectors can utilize such software in order to research and develop new ideas and topics. They categorize three areas of research: governance, organization, and innovation process. In addition, they took a look at motivations for contributions and competitive dynamics to showcase the complexity of research and how the open-source software can help to mitigate the difficulties that come with the research. With an open-market "laissez-faire" attitude, the article mentions how cost-cutting innovation can lead to competition that allows for greater inventions and more efficient management. Creativity is needed for individuals or companies to reduce cost, and open-source is an effective and free tool in order to obtain the datasets needed (Krogh and Hippel, 2006)

Another example of OSS being used can be found in the financial/business sector, where open source projects can be commoditized into something that piques self or corporate financial interest. Singh, Tan, and Mookerjee (2011) analyzed the impact of network social capital on open source projects and delved into the psychology of the individuals involved in the creation of such projects. Structural capital was inserted into the framework of OSS, which reflects its diversity of usage by any contributor. OSS being transformed into a tool for efficiency becomes a norm.

![image](https://user-images.githubusercontent.com/81675803/115168109-b0eb7600-a06e-11eb-8cd6-7563109a2946.png)
**Figure 1**

Open-source became a wonderful commodity that, theoretically, anyone can use. We see the pros of the articles reflected in our society today, as there are still artistic and important societal contributions being made while using OSS.

### Part 2: Closed-Source, More Efficiency
When conglomerates or companies make their own tool to use in-house or allow others to use their closed-source software (CSS) at a premium, it can be assumed that the software in itself must be effective enough for the whole conglomerate to utilize it at every level of management. With companies such as Google and Microsoft having their own CSS, they are able to ascertain something that OSS can never do: control. Whatever is a part of the network has every right to be utilized or discarded by these companies and, unlike the beginners and novices that you would typically find operating OSS, these companies have the capital in order to hire individuals that can push their CSS to consistently "state-of-the-art" rankings. The best and the brightest software developers, engineers, artists, philosophers, and contributors. Morality is ruled by the governing body of the country that these conglomerates reside in, but as far as content goes, these conglomerates have, more or less, every right to utilize the data in whatever way they best see fit.

The potentiation of CSS ultimately relies on the people that are working on it. The diversity of the source makers are dependent on the source makers themselves, which can either be good or bad depending on the hiring processes of the company. The data in itself also relies on the agenda of the company and to what ends does the company goes in order to obtain profits.

### Part 3: Open Anarchy: Good or Bad?
The definition of anarchy is that it is contrary to hierarchical structures, which is the political system in which political actors will centralize power to order to create spheres of authority and inferiority. This viewpoint, which is often taken by international relations theorists, expands the theory of anarchy to mean more than just disorder and chaos. The Westphalian model of anarchy is combined with the basic facets of OSS in order to assess human international interactions through a feasible modality in which researchers can grasp our internet relationships. David Fidler intertwines the processes of OSS with the historical evidence compiled from the annals of human history. Power and Ideas are two main concepts in which OSS propagates extremely effectively on the internet (Fidler, 2008).

The real questions with OSS are: Who governs the validity of the open-source data? Who gets the govern the dataset and why do they get to govern it? The question of OSS lies in its susceptibility to which truth can be muddled and there is a lack of governance. 

Although CSS has its regulatory affairs, closed-source data also has its issues as well. The conglomerate who is in charge of the data can do whatever they would like with it without the disclaimer of the individuals it has extracted from. The lack of open collaboration also means that there is no way for there to be as diverse as the collaborators there would be for OSS. CSS strips down privacy by selling off personal data to an institution that would give them profit and there are still security issues that come with CSS. CSS is imperfect as well, and this has led to many lawsuits that we see today.

Ultimately, with information that can come so easily, data can always be commoditized for one's interest.

### Part 4: Sources
Fidler, David P. "A Theory of Open-Source Anarchy." Indiana Journal of Global Legal Studies 15, no. 1 (2008): 259-84. Accessed April 19, 2021. doi:10.2979/gls.2008.15.1.259.

Singh, Param Vir, Yong Tan, and Vijay Mookerjee. "Network Effects: The Influence of Structural Capital on Open Source Project Success." MIS Quarterly 35, no. 4 (2011): 813-29. Accessed April 19, 2021. doi:10.2307/41409962.

Von Krogh, Georg, and Eric Von Hippel. "The Promise of Research on Open Source Software." Management Science 52, no. 7 (2006): 975-83. Accessed April 19, 2021. http://www.jstor.org/stable/20110574.

**Comment by epham36** [comment]
Thank you for bringing up that throughline between OSS and anarchy. I think it operates with an ethos that is often diametrically opposed to closed source software that feed into capitalistic systems of accumulation and production -- or at least can be used in that way. Definitely the questions you pose regarding who has control over this tool and ascertains the validity bring to mind the tragedy of the commons and the ability of people to collectively monitor the health of a shared resource. Whether a dictating/governing body is even needed is dictated by the ability of the collective to self-govern, distribute power, and extend agency.

**Comment by Wny-Duong** [comment]
(This is a bunch of stuff discussing the actual efficiency of open vs closed source code and it was both a pain to write and probably a pain to read, fun to look into though.)

In regards to Part 2 of your post, the efficiency/quality of open source vs closed source code is a very hotly debated topic and area of some research in the field of management. There was an analysis paper looking at code for various open and closed source operating systems (FreeBSD, Linux, Solaris [before its acqusition by Oracle], and Windows) [1]. Findings indicated that no matter what minor differences there may be (Linux had issues in code styles [stuff like a lack of unsafe function-like macros, %style conforming lines, etc.] but excelled in testing regarding code structure metrics and the opposite for Windows's operating kernel code), the structure and overall quality of the code will be very similar as long as you're comparing functionally similar software setting out to do the same thing, in this case, acting as the operating software. 

Interestingly, this paper does mention the effect of programming discipline in closed source projects as being a noticeable thing when it comes to reducing reliance on an example case of reducing C preprocessing [use of macros and preprocessor directives like #include, #define, etc.], which open source software seems to suffer from. The issue regarding preprocessing, at a completely high level of description because honestly I got way too in over my head in looking up papers for this, is that it helps facilitate real-world software development at the cost of maintainability, reliability, and portability of software.  

I'd imagine these things would be higher priority in a closed-source project simply because authority figures might bring certain priorities into the project. It's suggested that this practice of relying on preprocessing is problematic in long-term development, when a developer has incentive to work on a single project for a long amount of committed time (which is likelier in the case of a closed source project). Projects that would be more open source would feature assistance and contributions from people without a similar sort of commitment (note: There are open source developers who get paid for and are employed through their projects such as Chromium, Linux, etc.), which would enable the use of problematic programming practices to establish influence in their project.

These smaller differences in development practices however, they're not enough to differentiate the utility of similar open vs closed-source software to the consumer (in terms of performance, not in terms of feature-richness like if you were comparing GIMP to Photoshop or LibreOffice to Microsoft Office Suites). The more serious internal issues like security or stability of the software would be negligible in difference between open and closed source projects based off the paper I first cited.

This second paper that I wanted to bring up [2] has a ridiculous amount of math that goes into trying to quantify the quality of contributions to software development projects which is all very, very impressive and convoluted as Boelter Hall's layout, but it's also extremely informative. I'll just draw attention to the section regarding The Quality Debate on Open Source Software. There's reiteration of stuff we learned in class about the lower cost of open source software in deployment/operation. There's also the suggestion that the development process for open-source is actually much more efficient in terms of speed, with certain functionality in the Linux OS software being developed half a decade earlier than they were first seen in closed-source competitors like Windows [64-bit processing support in 1995 vs 2000, earlier adoption of IPv6 internet protocols]. The paper suggests this to likely be a result of feedback, suggestions, and dedicated forums to the development of these open source projects.

The perceived inefficiency from open source comes from a lack of authority, which leads into wasted time due to duplicated work, lack of attention to support documentation, and conformity to specifications. I don't think these are good ways to measure inefficiency, as a lot of these reasons for inefficiency are simply barriers to proper management and oversight of a project. Good management certainly might expediate the process of development under competent leadership and organization, but it's already established that certain open source software already see faster development in spite, or rather maybe because, of not having such authority.

[1] A Tale of Four Kernels https://www2.dmst.aueb.gr/dds/pubs/conf/2008-ICSE-4kernel/html/Spi08b.html

[2] Open Source Versus Closed Source: Software Quality in Monopoly and Competitive Markets https://www.researchgate.net/publication/3412475_Open_Source_Versus_Closed_Source_Software_Quality_in_Monopoly_and_Competitive_Markets/link/5727f57108aef9c00b8b527c/download
